front-packages/akeneo-design-system/src/components/Block/Block.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {isValidElement, ReactNode, Ref, SyntheticEvent} from 'react';
+import styled from 'styled-components';
+import {AkeneoThemedProps, getColor, getFontSize} from '../../theme';
+import {Override} from '../../shared';
+import {CloseIcon, IconProps} from '../../icons';
+import {IconButton} from '../IconButton/IconButton';
+
+type BlockProps = Override<
+  React.ButtonHTMLAttributes<HTMLButtonElement> & React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  {
+    /**
+     * Define if the block is removable.
+     */
+    removable?: boolean;
+
+    /**
+     * Function called when the user clicks on the remove button.
+     */
+    onRemove?: () => void;
+
+    /**
+     * Accessibility text for the remove icon button.
+     */
+    removeLabel?: string;
+
+    /**
+     * Accessibility label to describe shortly the button.
+     */
+    ariaLabel?: string;
+
+    /**
+     * Define which element is the label of this button for accessibility purposes. Expect a DOM node id.
+     */
+    ariaLabelledBy?: string;
+
+    /**
+     * Define what element is describing this button for accessibility purposes. Expect a DOM node id.
+     */
+    ariaDescribedBy?: string;
+
+    /**
+     * Children of the button.
+     */
+    children?: ReactNode;
+  }
+>;
+
+const ActionsContainer = styled.div`
+  height: 0;
+  display: none;
+  align-items: center;
+`;
+
+const Container = styled.div<AkeneoThemedProps>`
+  padding: 14px 20px;
+  border-style: solid;
+  border-width: 1px;
+  border-radius: 2px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  font-family: inherit;
+  font-size: ${getFontSize('default')};
+  font-weight: 400;
+  outline-style: none;
+  text-decoration: none;
+
+  background-color: ${getColor('white')};
+  border-color: ${getColor('grey', 80)};
+  color: ${getColor('grey', 140)};
+
+  &:hover {
+    background-color: ${getColor('grey', 20)};
+    ${ActionsContainer} {
+      display: flex;
+    }
+  }
+`;
+
+const Block = React.forwardRef<HTMLButtonElement, BlockProps>(
+  (
+    {
+      removable = false,
+      onRemove,
+      removeLabel,
+      ariaDescribedBy,
+      ariaLabel,
+      ariaLabelledBy,
+      children,
+      ...rest
+    }: BlockProps,
+    forwardedRef: Ref<HTMLButtonElement>
+  ) => {
+    const handleRemove = (event: SyntheticEvent) => {
+      event.preventDefault();
+      if (onRemove) {
+        onRemove();
+      }
```

Comment:
```suggestion
      removable && onRemove();
```

And if it doesn't work 
```suggestion
      removable && onRemove.?();
```